test/math.jl

```diff
@@ -678,3 +678,20 @@ let x = 2.0
     @test exp2(Float22716(x)) === 2^x
     @test exp10(Float22716(x)) === 10^x
 end
+
+@testset "asin #23088" begin
+    for T in (Float32, Float64)
+        @test asin(zero(T)) == zero(T)
+        @test asin(-zero(T)) == -zero(T)
+        @test asin(nextfloat(zero(T))) == nextfloat(zero(T))
+        @test asin(prevfloat(zero(T))) == prevfloat(zero(T))
+        @test asin(one(T)) == T(pi)/2
+        @test asin(-one(T)) == -T(pi)/2
+        for x in (0.45, 0.6, 0.98)
+            by = T(asin(big(x)))
```

Comment:
If you add 0.9999 to the list of numbers to test, the current way to test would report that the error is zero, while it's actually 4 ulp if you use the correct way to convert to `BigFloat`:
```julia
julia> x = 0.9999
0.9999

julia> by = Float64(asin(big(x)))
1.5566540733173846

julia> asin(x)
1.5566540733173846

julia> abs(asin(x) - by) / eps(by)
0.0

julia> By = Float64(asin(parse(BigFloat, repr(x))))
1.5566540733173837

julia> abs(asin(x) - By) / eps(by)
4.0
```
The error growths as `x` approaches 1, up to 3.418348e6 ulp for `x = 0.9999999999999999`.

However,
```julia
julia> sin(1.5566540733173846)
0.9999

julia> sin(1.5566540733173837)
0.9999
```
so both numbers are mapped to `0.9999` by `sin`.

It looks like that an accurate docstring of `asin` would be

> Compute a number that is mapped into `x` by `sin`

rather than

> Compute the inverse sine of `x`

This has nothing to do with the proposed implementation of the function, also the previous one gives the same results.